include/kframework/foundry.md

```diff
@@ -558,9 +559,12 @@ Ignore all cheat code calls which take place while `expectRevert` is active.
       [priority(35)]
 ```
 
+The `#halt` production is used to examine the end of each call in KEVM.
+If the call depth of the current call is lower than the call depth of the `expectRevert` cheat code and the `<statusCode>` is not `EVMC_SUCCESS`, the `#checkRevertReason` will be used to compare the output of the call with the expect reason provided.
+
 ```{.k .bytes}
     rule [foundry.handleExpectRevert]:
-         <k> (. => #checkRevertReason ~> #clearExpectRevert) ~> #halt ... </k>
+         <k> (. => #encodeOutput ~> #checkRevertReason ~> #clearExpectRevert) ~> #halt ... </k>
          <statusCode> SC </statusCode>
          <callDepth> CD </callDepth>
          <expectedRevert>
```

Comment:
I think instead of having a whole `#encodeOutput` thing, since it's _only_ used here, let's jsut make it a function and inline it into this rule?

Reply:
I'd also rather not modify the contents of the `<output>` cell directly like `<encodeOutput>` does, because anything that comes _after_ this step which expects the `<output>` cell to be unencoded will break. So I think it's best we just call the function to do the encoding at the time we're comparing the outptu against the expected value.

Reply:
Done!